gguf-py/gguf/constants.py

```diff
@@ -181,6 +182,7 @@ class MODEL_TENSOR(IntEnum):
     SSM_A              = auto()
     SSM_D              = auto()
     SSM_OUT            = auto()
+    FFN_NORM_EXP       = auto()
```

Comment:
Since the actual numbers associated to the enum values of `MODEL_TENSOR` don't really matter (their names (from `TENSOR_NAMES`) are used instead in GGUF), maybe `FFN_NORM_EXP` could be placed right before `FFN_GATE_EXP`, a bit like `FFN_NORM` is right before `FFN_GATE`, for consistency.

If this is changed, it should also be placed similarly in `TENSOR_NAMES` and `MODEL_TENSORS[MODEL.ARCTIC]` in `gguf-py/gguf/constants.py` as well as in the `llm_tensor` enum, the `LLM_TENSOR_NAMES` mapping, and the `llama_layer` struct (and maybe the `LLM_ARCH_ARCTIC` case in `llm_load_tensors`?) in `llama.cpp`.

Reply:
I changed the order as requested, but in llama_layer struct the order is different, so I didn't touch it. In llm_load_tensors I think it was already in the requested order.